src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs

```diff
@@ -69,7 +69,7 @@ public Task ConnectAsync(Uri uri, CancellationToken cancellationToken)
             switch ((InternalState)Interlocked.CompareExchange(ref _state, (int)InternalState.Connecting, (int)InternalState.Created))
             {
                 case InternalState.Disposed:
-                    throw new ObjectDisposedException(GetType().FullName);
+                    ObjectDisposedException.Throw(this);
```

Comment:
Probably simplest just to revert this line rather than trying to work around the compiler not knowing that Throw never returns.